src/AggregateFunctions/AggregateFunctionSum.h

```diff
@@ -457,13 +457,12 @@ class AggregateFunctionSum final : public IAggregateFunctionDataHelper<Data, Agg
 
     String getName() const override
     {
-        if constexpr (Type == AggregateFunctionTypeSum)
-            return "sum";
-        else if constexpr (Type == AggregateFunctionTypeSumWithOverflow)
-            return "sumWithOverflow";
-        else if constexpr (Type == AggregateFunctionTypeSumKahan)
-            return "sumKahan";
-        UNREACHABLE();
+        switch (Type)
```

Comment:
I would leave it as is, but remove the `UNREACHABLE()` line. clang is [smart](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:22,endLineNumber:18,positionColumn:22,positionLineNumber:18,selectionStartColumn:22,selectionStartLineNumber:18,startColumn:22,startLineNumber:18),source:'%23include+%3Ciostream%3E%0A%23include+%3Cstring%3E%0A%23include+%3Calgorithm%3E%0A%23include+%3Cfunctional%3E%0A%23include+%3Coptional%3E%0A%0Aenum+AggregateFunctionSumType%0A%7B%0A++++AggregateFunctionTypeSum,%0A++++AggregateFunctionTypeSumWithOverflow,%0A++++AggregateFunctionTypeSumKahan,%0A++++SomeExtra,%0A%7D%3B%0A%0Atemplate+%3CAggregateFunctionSumType+Type%3E%0Astd::string+getValue()+%7B%0A++++if+constexpr+(Type+%3D%3D+AggregateFunctionTypeSum)%0A++++++++return+%22sum%22%3B%0A++++else+if+constexpr+(Type+%3D%3D+AggregateFunctionTypeSumWithOverflow)%0A++++++++return+%22sumWithOverflow%22%3B%0A++++else+if+constexpr+(Type+%3D%3D+AggregateFunctionTypeSumKahan)%0A++++++++return+%22sumKahan%22%3B%0A%7D%0A%0Aint+main(int+argc,+char+**+argv)+%7B%0A++++return+getValue%3CSomeExtra%3E().length()%3B%0A%0A%7D'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0')),k:100,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+clang+(trunk)+(Compiler+%231)',t:'0'),(h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'0',intel:'0',libraryCode:'1',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-O3+-std%3Dc%2B%2B20+-stdlib%3Dlibstdc%2B%2B',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),l:'3',n:'0',o:'',t:'0')),version:4) enough to detect that a non-void function doesn't return a value in all path.